Services/CAS/classes/class.ilAuthProviderCAS.php

```diff
@@ -121,17 +125,32 @@ public function doAuthentication(\ilAuthStatus $status)
 	protected function handleLDAPDataSource(\ilAuthStatus $status)
 	{
 		include_once './Services/LDAP/classes/class.ilLDAPServer.php';
+		// Fixed : AUTH_CAS -> AUTH_LDAP
 		$server = ilLDAPServer::getInstanceByServerId(
-			ilLDAPServer::getDataSource(AUTH_CAS)
+			ilLDAPServer::getDataSource(AUTH_LDAP)
```

Comment:
This modification shouldn't be necessary. 
Maybe this works in your case but the ldap server should be configured as datasource for the authentication mode "CAS".